packages/playwright/src/index.ts

```diff
@@ -701,7 +692,75 @@ class ArtifactsRecorder {
     })));
 
     await this._screenshotRecorder.persistTemporary();
-    await this._pageSnapshotRecorder.persistTemporary();
+    if (!process.env.PLAYWRIGHT_NO_COPY_PROMPT && leftoverContexts.length > 0)
+      await this._attachErrorPrompts(leftoverContexts[0]);
+  }
+
+  private async _attachErrorPrompts(context: BrowserContext) {
+    if (this._testInfo.errors.length === 0)
+      return;
+
+    let pageSnapshot: string | undefined;
+    const page = context.pages()[0];
+    if (!page)
+      return;
+
+    try {
+      pageSnapshot = await page.locator('body').ariaSnapshot({ timeout: 5000 });  // TODO: maybe capture snapshot when the error is created, so it's from the right page and right time
+    } catch {}
+
+    const testSources = await fs.promises.readFile(this._testInfo.file, 'utf-8');
+    for (const [index, error] of this._testInfo.errors.entries()) {
+      const metadata = this._testInfo.config.metadata as MetadataWithCommitInfo;
+
+      const promptParts = [
+        `My Playwright test failed.`,
+        `Explain why, be concise, respect Playwright best practices.`,
+        '',
+        `Failed test: ${this._testInfo.titlePath.join(' >> ')}`,
+        '',
+        'Error:',
+        '',
+        '```js',
```

Comment:
Why is this a `js` snippet? Shouldn't it be plaintext or no language indication?

Reply:
you're right, it was `js` before because it contained the snippet, and the stacktrace was JS-style. i'll remove the language indication